src/sanitizer.js

```diff
@@ -8,18 +8,67 @@
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocFragment = function _sanitizeDocFragment(config, input) {
     // TODO implement this function and remove eslint-disable comment above
+    // TODO maybe we can use https://developer.mozilla.org/en-US/docs/Web/API/NodeIterator to traverse through nodes
 };
 
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocument = function _sanitizeDocument(config, input) {
     // TODO implement this function and remove eslint-disable comment above
 };
 
-// a parser that doesn't run scripts or loads resources
-// eslint-disable-next-line no-unused-vars
+/**
+ * fragmentParser
+ * a parser that doesn't run scripts or loads resources
+ *
+ * @param {Node} ctx DOM node
+ * @param {Object} input Document or DocumentFragment
+ */
+// Q: is the purpose of _fragmentParser is to parse `input` under the context of `ctx`, so that `input` can be append to `ctx` correctly and returned?
```

Comment:
`fragmentParser` is where all HTML parsing is supposed to happen. i.e., everywhere the spec says "invoke the fragment parsing algorithm", we call this function. This algorithm is a core algorithm of HTML parsing in the spec, see https://html.spec.whatwg.org/multipage/parsing.html#html-fragment-parsing-algorithm


The function parameters are a DOM node and an HTML string.
I thought the output should always be a Document Fragment, but it reads to me from the HTML spec that we'd return the top-most node from the fragment and its child as a subtree. Not sure what the difference is?
Probably easiest to kick it off with whatever is easiest and file a follow-up issue.

Reply:
Got it! I'll make sure to file a issue to look into this if we end up committing a `_fragmentParser` that returns a `DocumentFragment`